### PR TITLE
test: Use MiniWallet in test_no_inherited_signaling RBF test

### PR DESCRIPTION
**Pull request by maflcko**
This comes with nice benefits:
* Less code and complexity
* Test can be run without wallet compiled in

Also add some additional checks for `getmempoolentry` (#22209) and other cleanups :art:

### CODE REVIEW

**Comment by theStack** [comment]
Concept ACK

Not strictly related to this PR, but: the parameter `locktime` is currently unused in `send_self_transfer`, i.e. it is not passed on to `create_self_transfer`. Now that this part is touched in this PR, this could also be addressed (e.g. in the first commit).

File: test/functional/feature_rbf.py [hunk 1]
```diff
@@ -550,86 +550,80 @@ def test_prioritised_transactions(self):
...
         assert_equal(json0["vin"][0]["sequence"], 4294967293)
         assert_equal(json1["vin"][0]["sequence"], 4294967294)
 
     def test_no_inherited_signaling(self):
-        # Send tx from which to conflict outputs later
-        base_txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), Decimal("10"))
-        self.nodes[0].generate(1)
-        self.sync_blocks()
+        wallet = MiniWallet(self.nodes[0])
+        wallet.scan_blocks(start=76, num=1)
```
Comment (mjdietzx) [inline test/functional/feature_rbf.py:572]: Where does the hardcoded `76` come from? It seems to me you just need a mature utxo here? If that's the case and I'm not missing anything, could you do something like: `start=blockheight - COINBASE_MATURITY`?

File: test/functional/feature_rbf.py [hunk 2]
```diff
@@ -78,10 +81,7 @@ def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
```
Comment (mjdietzx) [inline test/functional/feature_rbf.py:81]: Did you forget to remove this now that the test can be run without the wallet compiled?

File: test/functional/feature_rbf.py [hunk 3]
```diff
@@ -550,86 +550,80 @@ def test_prioritised_transactions(self):
...
+            mempool_valid=False,
+        )
 
         # Broadcast replacement child tx
         # BIP 125 :
         # 1. The original transactions signal replaceability explicitly or through inheritance as described in the above
         # Summary section.
-        # The original transaction (`optout_child_tx`) doesn't signal RBF but its parent (`optin_parent_txid`) does.
+        # The original transaction (`optout_child_tx`) doesn't signal RBF but its parent (`optin_parent_tx`) does.
         # The replacement transaction (`replacement_child_tx`) should be able to replace the original transaction.
```
Comment (mjdietzx) [inline test/functional/feature_rbf.py:622]: Is my understanding here correct?

a) `testmempoolaccept` returns that `replacement_child_tx` is allowed even though it is rejected w/ 'txn-mempool-conflict'
b) `getmempoolentry` returns that `optout_child_tx` is 'bip125-replaceable' - as you descbribe in https://github.com/bitcoin/bitcoin/issues/22209 even though it is not

So does anything need to be done to fix a and b?

Another question I have (which maybe I should just try out). But, I wonder what would happen if you tried something like https://github.com/bitcoin/bitcoin/pull/22210/files#diff-260c479178e3e99a16798aeed39f400676894392864784eda6c207716704398dR584-R589 after `optout_child_tx` was already sent

**Comment by practicalswift** [comment]
Concept ACK: `MiniWallet` good

File: test/functional/feature_rbf.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/feature_rbf.py:572]: The constant is derived from https://github.com/bitcoin/bitcoin/blob/96f828ba4d69f8acadf335ee79fa8f06cd33c0a6/test/functional/test_framework/test_framework.py#L736-L748

File: test/functional/feature_rbf.py (hunk 2 above)
Comment (MarcoFalke) [reply test/functional/feature_rbf.py:81]: Can it?

File: test/functional/feature_rbf.py (hunk 3 above)
Comment (MarcoFalke) [reply test/functional/feature_rbf.py:622]: a) No, it is also rejected by testmempoolaccept, see `mempool_valid=False` for `replacement_child_tx` above.
b) correct

> So does anything need to be done to fix a and b?

Maybe. But a fix should be done outside a test-only change.

File: test/functional/feature_rbf.py (hunk 2 above)
Comment (mjdietzx) [reply test/functional/feature_rbf.py:81]: Nope 👍

**Comment by mjdietzx** [comment]
Tested ACK fa7d71f270b89c9d06230d4ff262646f9ea29f4a thanks for the explanations, nicely done

**Review by theStack — APPROVED** [review]
ACK fa7d71f270b89c9d06230d4ff262646f9ea29f4a 🍷 

Follow-up idea based on discussion https://github.com/bitcoin/bitcoin/pull/22210#discussion_r650178263: This will probably not the last test that needs the block number of the first output to `ADDRESS_BCRT1_P2WSH_OP_TRUE` (i.e. suitable for MiniWallet in its default mode of `MiniWalletMode.ADDRESS_OP_TRUE`). In the future it may makes sense to put the magic number 76 into a constant.

**Pull request merged**